docs/certificate-specification.md

```diff
@@ -0,0 +1,125 @@
+* Specify other Extended Key Usages besides Code Signing
+
+An intermediate certificate SHOULD:
+
+* Specify a Basic Constraints path length constraint of 0, `pathlen:0`. This limits the intermediate
+  CA to only issue end-entity certificates
+* Use the signing algorithm ECDSA NIST P-384 (secp384r1) or stronger, or RSA-4096
+* Have a lifetime that does not require frequent rotation, such as 3 years
+
+An intermediate certificate SHOULD NOT:
+
+* Use a different signature scheme (ECDSA vs RSA) than its parent certificate, as some clients do not support this
+
+An intermediate certificate MAY:
+
+* Be optional. An end-entity certificate MAY be issued from a root certificate or an intermediate certificate.
+  Clients MUST be able to verify a chain with any number of intermediate certificates.
+
+## Issued Certificate
+
+An issued certificate MUST:
+
+* Specify exactly one Subject Alternative Name, as a critical extension. It MUST be populated by either:
+   * An email
+   * A URI
+* Specify an Issuer equal to the parent certificate's Subject
+* Specify a Key Usage for Digital Signature 
+* Specify an Extended Key Usage for Code Signing
+* Specify a lifetime that does not exceed the parent certificiate
+* Specify a unique, positive, 160 bit serial number
+* Specify a Subject Key Identifier
+* Specify an Authority Key Identifier equal to the parent certificate's Subject Key Identifier 
+* Specify an empty Subject
+* Be compliant with [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)
+* Specify a public key that is either:
+   * ECDSA NIST P-256, NIST P-384, or NIST P-521
+   * RSA of key size 2048 to 4096 (inclusive) with size % 8 = 0, E = 65537, and containing no weak primes
+   * ED25519
+* Specify the OpenID Connect identity token issuer with OID `1.3.6.1.4.1.57264.1.1`
+* Be appended to a Certificate Transparency log. Clients MUST NOT trust certificates that do not present
+  either a proof of inclusion or a Signed Certificate Timestamp (SCT)
+
+An issued certificate MUST NOT:
+
+* Specify a nonempty Subject
+* Specify multiple Subject Alternative Name values
+* Specify other Key Usages besides Digital Signature
+* Specify other Extended Key Usages besides Code Signing
+
+An issued certificate SHOULD:
+
+* Use an ephemeral key. A client MAY request a certificate with a long-lived key, but a client MUST
+  adequately secure the key material
+* Append a precertificate to a Certificate Transparency log, where the precertificate MUST be signed by the certificate authority
+  and MUST include a poison extension with OID `1.3.6.1.4.1.11129.2.4.3`
+* Specify the Signed Certificate Timestamp (SCT) from the Certificate Transparency log with OID `1.3.6.1.4.1.11129.2.4.2`
+
+An issued certificate SHOULD NOT:
+
+* Use a different public key scheme (ECDSA vs RSA) than its parent certificate, as some clients do not support this
+* Specify a public key that is stronger than its parent certificate
```

Comment:
I might be missing it, but this is a little unintuitive to me: why should the ephemeral signing certificate be limited in key strength to no more than its issuer?

Reply:
The thought is that as keys fail due to new weaknesses, the issued certificates should be compromised before the parent certificate. If a parent's key is weaker and it is compromised, then it can simply issue new children. If a child's key is weaker, then there's no impact on the parent if it's compromised.

Reply:
Makes sense!